engine/consensus/message_hub/message_hub.go

```diff
@@ -302,10 +293,14 @@ func (h *MessageHub) processQueuedVote(packed *packedVote) error {
 	return nil
 }
 
-// processQueuedProposal performs actual processing of flow.Header, as a result of successful invocation
-// broadcasts block proposal to consensus committee and rest of Flow network.
+// sendOwnProposal propagates the block proposal to the consensus cluster and submits to non-consensus network:
+//   - directly forwarded proposal to HotStuff core logic
+//     (skipping compliance engine as we assume our own proposals to be correct)
```

Comment:
would need to update this comment, if we move the logic for forwarding the proposal to hotstuff and the vote aggregator.